src/audio/drc/drc_math_hifi3.c

```diff
@@ -62,35 +55,51 @@ static inline ae_f32 log10_fixed(ae_f32 x)
 	 * fpminimax(log10(x), 5, [|SG...|], [1/2;sqrt(2)/2], absolute);
 	 * max err ~= 6.088e-8
 	 */
-	int32_t e;
+	int bit = 31 - AE_NSAZ32_L(x);
+	int32_t e = bit - 26;
 	int32_t lshift;
 	ae_f32 exp; /* Q7.25 */
 	ae_f32 acc; /* Q6.26 */
 	ae_f32 tmp; /* Q6.26 */
+	ae_f64 tmp64;
 
-	x = rexp_fixed(x, 26, &e); /* Q2.30 */
+	x = AE_SRAA32(x, bit - 30);
 	exp = e << 25; /* Q_CONVERT_FLOAT(e, 25) */
 
 	if ((int32_t)x > (int32_t)ONE_OVER_SQRT2_Q30) {
-		lshift = drc_get_lshift(30, 30, 30);
-		x = drc_mult_lshift(x, ONE_OVER_SQRT2_Q30, lshift);
+		tmp64 = AE_MULF32R_LL(x, ONE_OVER_SQRT2_Q30);
+		/* SHIFT_IDX_QX30_QY30_QZ30 = 1 */
+		lshift = SHIFT_IDX_QX30_QY30_QZ30;
+		tmp64 = AE_SLAA64S(tmp64, lshift);
+		x = AE_ROUND32F48SSYM(tmp64);
 		exp = AE_ADD32(exp, HALF_Q25);
 	}
 
-	lshift = drc_get_lshift(26, 30, 26);
-	acc = drc_mult_lshift(LOG10_FUNC_A5_Q26, x, lshift);
+	tmp64 = AE_MULF32R_LL(LOG10_FUNC_A5_Q26, x);
+	lshift = SHIFT_IDX_QX26_QY30_QZ26;
+	tmp64 = AE_SLAA64S(tmp64, lshift);
```

Comment:
Use AE_SLAI64S() to all constant shift values, you can also eliminate lshift variable.